src/fastga/models/weight/mass_breakdown/b_propulsion/b1_engine_weight.py

```diff
@@ -50,7 +50,7 @@ def setup(self):
 
         self.add_output("data:weight:propulsion:engine:mass", units="lb")
 
-        self.declare_partials("*", "*", method="fd")
+        self.declare_partials(of="*", wrt="*", method="exact")
```

Comment:
You actually don't know the exact derivative for all the variables here. Let me explain, the propulsion wrapper adds a bunch of inputs that he uses for some computation on which we have no control here so the only variable which you really have a control on is the number of engine